### DISCUSSION TITLE
Wireguard Integration

### DISCUSSION THREAD

**Discussion by cdemkos** [raspiblitz/raspiblitz#4828] [Feature / App Ideas]
Since the integration of Tailscale in the Version 1.11.3 i would welcome to see an similar integration of Wireguard for user with an existing Server infrastructure. Thiis would allow to have a choice with selection of the VPN Service

**Comment by rootzoll** [comment]
Can you describe more clearly how you would use wireguard here with RaspiBlitz?

Normally RaspiBlitz is already part of your local home network. If you run a seperated wire guard server (on your router or proxmox) you would connect with your phone to that server - now your phone is part of the local home network and can connect to RaspiBlitz services.

Are you asking to add a Wireguard server to RaspiBlitz?

**Comment by cdemkos** [comment]
Raspiblitz should be the client connected to the server
Im aware that this is usualy solved by a config file. But i think it would be neat to have an entry for that in the menu .

**Comment by rootzoll** [comment]
> Raspiblitz should be the client connected to the server

I dont get it yet - what would be the benefit? I use Wireguard when I have a laptop or mobile phone (clients) on the road but want them to be part of my local LAN to access my local services - like the RaspiBlitz. The RaspiBlitz is not a mobile node - its at my home and already naturally part of my local LAN .. why should I run a Wireguard client on RaspiBlitz and connect to a server?

@cdemkos please give me a clear picture of the use case

**Comment by cdemkos** [comment]
Yes i understand your point  with the local network. The motivation is to have valid certificates on the raspiblitz. By now it is just possible to have self signed certificates.

**Comment by rootzoll** [comment]
> The motivation is to have valid certificates on the raspiblitz. 

How would I use Wireguard (RaspiBlitz as client) to get valid certificates?

**Comment by rootzoll** [comment]
Ah, okay! In that context, it makes sense. So you're referring to renting a small virtual server (VS) and using its public IP and domain. Now it’s clearer which setup you're considering.

**Comment by rootzoll** [comment]
Issue created for v1.13.0: #4832
